tools/tidb-binlog: update binlog deploy info (#1625)
pingcap/docs: dev/reference/tools/tidb-binlog/deploy.md

    Run the following command to use `binlogctl` to generate the `tso` information which is needed for the initial start of Drainer:

    {{< copyable "shell-regular" >}}

    ```bash
    cd /home/tidb/tidb-ansible
    ```

    ```

>     resources/bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd generate_meta

Reviewer: Should this line be moved to last block?

        -c int
            the number of the concurrency of the downstream for replication. The bigger the value, the better throughput performance of the concurrency (1 by default).
        -cache-binlog-count int
            the limit on the number of binlog items in the cache (512 by default)

>             If a single binlog in the upstream causes OOM in Drainer, try reduce the number of binlog items to reduce memory usage.

Reviewer: Maybe `item` is better than `file` here?

>             If a single binlog in the upstream causes OOM in Drainer, try reduce the number of binlog items to reduce memory usage.

Reviewer 2: Yes.

> - Make sure that the TiDB Binlog service is enabled in all TiDB instances in a same cluster, otherwise upstream and downstream data inconsistency might occur during data replication. If you want to temporarily run a TiDB instance where the TiDB Binlog service is not enabled, set `run_ddl=false` in the TiDB configuration file.
> - Drainer does not support the `rename` DDL operation on the table of `ignore schemas` (the schemas in the filter list).
> - If you want to start Drainer in an existing TiDB cluster, generally you need to make a full backup of the cluster data, obtain `savepoint`, import the data to the target database, and then start Drainer to replicate the incremental data from `savepoint`.

> > - When the downstream database is TiDB or MySQL, the  `sql_mode` in both upstream and downstream database should be consistent. In other words, the `sql_mode` should be the same when each SQL statement is executed in the upstream and synchronized in the downstream. You can execute `select @@sql_mode;` statement in upstream and downstream respectively to compare `sql_mode`.

Reviewer 2: ```suggestion
> - When the downstream database is TiDB or MySQL, ensure that the `sql_mode` in the upstream and downstream databases are consistent. In other words, the `sql_mode` should be the same when each SQL statement is executed in the upstream and replicated to the downstream. You can execute the `select @@sql_mode;` statement in the upstream and downstream respectively to compare `sql_mode`.
```

> > - Drainer also fails to synchronize when there are DDL statements that upstream database supports but downstream database does not. An example is synchronizing `CREATE TABLE t1(a INT) ROW_FORMAT=FIXED;` statement when the downstream database MySQL uses the InnoDB engine. In this case, you can configure [skipping transactions](/dev/reference/tools/tidb-binlog/faq.md#what-can-i-do-when-some-ddl-statements-supported-by-the-upstream-database-cause-error-when-executed-in-the-downstream-database) in Drainer, and manually execute compatible statements in the downstream database.

Reviewer 2: ```suggestion
> - When a DDL statement is supported in the upstream but incompatible with the downstream, Drainer fails to replicate data. An example is to replicate the `CREATE TABLE t1(a INT) ROW_FORMAT=FIXED;` statement when the downstream MySQL uses the InnoDB engine. In this case, you can configure [skipping transactions](/dev/reference/tools/tidb-binlog/faq.md#what-can-i-do-when-some-ddl-statements-supported-by-the-upstream-database-cause-error-when-executed-in-the-downstream-database) in Drainer, and manually execute the compatible statement in the downstream database.
```
